packages/runtime/src/runtime/entry.tsx

```diff
@@ -20,24 +21,32 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
 
   const hasMenu = "items" in entry;
   useEffect(() => {
-    if (screen && location.pathname !== `/${screen.name.toLowerCase()}`) {
-      navigate({
-        pathname: `/${screen.name.toLowerCase()}`,
-        search: location.search,
-      });
+    if (!keys(entry).length) {
       return;
     }
 
-    if (!(hasMenu && location.pathname === "/")) {
+    const navigateToPath = (pathname: string, search = ""): void => {
+      navigate({ pathname, search });
+    };
+
+    const getLowerCasePath = (name: string): string => `/${name.toLowerCase()}`;
+
+    const currentPath = location.pathname;
+
+    if (screen && currentPath !== getLowerCasePath(screen.name)) {
+      navigateToPath(getLowerCasePath(screen.name), location.search);
       return;
     }
 
-    const selectedItem =
-      entry.items.find((item) => item.selected) ?? entry.items[0];
-    navigate({
-      pathname: `/${selectedItem.page.toLowerCase()}`,
-      search: location.search,
-    });
+    if (hasMenu) {
+      if (currentPath === "/") {
+        const selectedItem =
+          entry.items.find((item) => item.selected) ?? entry.items[0];
+        navigateToPath(getLowerCasePath(selectedItem.page), location.search);
+      }
+    } else if (currentPath === "/") {
+      navigateToPath(getLowerCasePath(entry.name));
+    }
```

Comment:
I think you keep the check for selectedItem && menu, and then lastly navigate to home...so the priority should be like

if (path is not root)

if (screen)

if (hasMenu && selectedItem)

goHome

it's not really elseif it's just a series of conditions and can return early if met.